transport/src/main/kotlin/com/izivia/ocpi/toolkit/transport/TransportServer.kt

```diff
@@ -31,4 +31,9 @@ interface TransportServer {
      * Stops the server
      */
     fun stop()
+
+    /**
+     * The configured baseUrl this server listens to, used to generate NextPageUrl and Endpoint Url
+     */
+    fun baseUrl(): String
```

Comment:
In the lib, this field is only used by `registerOn` of `OcpiSelfRegisteringModuleServer` to specify the path of the endpoint of the module (`Endpoint Url`).

According to your `TransportServer` implementation, it can, indeed be the `baseUrl` used to generate `NextPageUrl` (as long as you pass it in the `baseUrl` field of the `HttpRequest`).

---

In our case, we have a multi tenant platform. So each platform can have its own `baseUrl` to contact us. So we dynamically compute the `baseUrl` every time we receive a request. In our case, we will not use `baseUrl()` and we will not use automatic endpoint registration because each tenant has its set of endpoints it can access.

---

Knowing that, I would like to change the documentation to say something like:
> The configured baseUrl this server listens to. <b>ONLY</b> used by the lib to generate Endpoint Url. You can use it in `handle` to specify the `baseUrl` of `HttpRequest`. In that case, it will be used to generate `NextPageUrl`

Then, in the lib, we have to make sure that this `baseUrl` is not used anywhere else (the comment implies that it is not). If a baseUrl has to be used, it will be the one in `HttpRequest`

We can even rename it to `staticBaseUrl()`, but I am not sure about that